kernel/init.c

```diff
@@ -165,7 +165,7 @@ void __weak z_early_memcpy(void *dst, const void *src, size_t n)
 __boot_func
 void z_bss_zero(void)
 {
-	if (IS_ENABLED(CONFIG_ARCH_POSIX)) {
+	if (IS_ENABLED(CONFIG_ARCH_POSIX) || IS_ENABLED(CONFIG_NO_BSS_INIT)) {
```

Comment:
Would be cleaner to just select/set this from the posix layer than have the parallel test and big comment.  Posix can set NO_BSS_INIT=y on its own and document it however it likes, people reading the kernel source shouldn't necessarily need to care, beyond "some platforms already did this".

Reply:
Ok, nice idea! Let's select it for POSIX, thanks!